omod/src/main/java/org/openmrs/module/dataintegrity/web/rest/resource/IntegrityRuleCategoryResource.java

```diff
@@ -0,0 +1,104 @@
+package org.openmrs.module.dataintegrity.web.rest.resource;
+
+import org.openmrs.api.context.Context;
+import org.openmrs.module.dataintegrity.DataIntegrityRule;
+import org.openmrs.module.dataintegrity.api.DataIntegrityService;
+import org.openmrs.module.dataintegrity.web.rest.controller.DataIntegrityRestController;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.PageableResult;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+@Resource(name = RestConstants.VERSION_1  + DataIntegrityRestController.DATA_INTEGRITY_REST_NAMESPACE + "/ruleCategory", supportedClass = DataIntegrityRule.class, supportedOpenmrsVersions = { "1.8.*",
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
+public class IntegrityRuleCategoryResource extends DelegatingSubResource<DataIntegrityRule, DataIntegrityRule, IntegrityRuleResource> {
+    @Override
+    public DataIntegrityRule getByUniqueId(String id) {
+        throw new ResourceDoesNotSupportOperationException();
+    }
+
+    @Override
+    protected void delete(DataIntegrityRule dataIntegrityRule, String s, RequestContext requestContext) throws ResponseException {
+        throw new ResourceDoesNotSupportOperationException();
+    }
+
+    @Override
+    public DataIntegrityRule newDelegate() {
+        return new DataIntegrityRule();
+    }
+
+    @Override
+    public DataIntegrityRule save(DataIntegrityRule dataIntegrityRule) {
+        throw new ResourceDoesNotSupportOperationException();
```

Comment:
Why we can't create new DataIntegrityRule using the REST?

Reply:
We can. It is yet to be implemented.